pkg/storer/internal/reserve/reserve.go

```diff
@@ -335,6 +360,14 @@ func (r *Reserve) EvictBatchBin(
 		return 0, err
 	}
 
+	// evict oldest chunks first
+	sort.Slice(evicted, func(i, j int) bool {
+		return evicted[i].BinID < evicted[j].BinID
```

Comment:
Color me not fully grokking this, but how is sorting by BinID getting us the "oldest chunks first" as the comment states?

Reply:
binID is an unique, incremented value that is assigned to each chunk in the reserve. So by removing the chunks with the lowest binID, we can at least bring some order (and not just random) as to which chunks should be removed first.

Reply:
So binID has nothing to do with the pullsync bins, nor anything else within bee that is called bin?   And since it is unique, I would gather that each "bin" contains but a single chunk?  Bizarre.

One of my personal software development mantras is "Naming can make or break a project".

Reply:
the reserve has "bins", which correspond to the proximity orders of the chunks (bin 0 = po 0, etc). so a "binID" is the unique ID of a chunk in the bin.

pullsync gets chunks from a peer using the peer's binIDs (so called bin intervals) for each bin/po.